generate.go

```diff
@@ -4,5 +4,6 @@
 //go:generate go run github.com/acorn-io/baaah/cmd/deepcopy ./pkg/apis/admin.acorn.io/v1/
 //go:generate go run k8s.io/kube-openapi/cmd/openapi-gen -i github.com/acorn-io/acorn/pkg/apis/internal.acorn.io/v1,github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1,github.com/acorn-io/acorn/pkg/apis/internal.admin.acorn.io/v1,github.com/acorn-io/acorn/pkg/apis/admin.acorn.io/v1,k8s.io/apimachinery/pkg/apis/meta/v1,k8s.io/apimachinery/pkg/runtime,k8s.io/apimachinery/pkg/version,k8s.io/apimachinery/pkg/api/resource,k8s.io/api/core/v1,k8s.io/api/rbac/v1 -p ./pkg/openapi/generated -h tools/header.txt
 //go:generate go run github.com/golang/mock/mockgen --build_flags=--mod=mod -destination=./pkg/mocks/mock_client.go -package=mocks github.com/acorn-io/acorn/pkg/client Client,ProjectClientFactory
+//go:generate go run github.com/golang/mock/mockgen --build_flags=--mod=mod -destination=./pkg/mocks/k8s/mock.go -package=mocks sigs.k8s.io/controller-runtime/pkg/client Reader
```

Comment:
Another option I used when mocking the controller was to use the k8's fake package for controller-runtime: sigs.k8s.io/controller-runtime/pkg/client/fake. This was a more finicky to figure out how to use our CRD's, but I assumed the fake library would be the most hardened way to do that.

Good example at acorn/pkg/client/info_test.go

I also like this approach though and don't see anything wrong with it

Used this in acorn/pkg/client/info_test.go

Reply:
yeah, I started there as well. Went with this mock for a few reasons:
1. see how hard/easy it would be to do
2. I didn't really want to depend on the fake client's behavior if a standard mock would do. Seemed like having a single approach to managing mocks made sense.

Reply:
Yeah, this is way more straightforward than using fake.